src/Orleans/Streams/SimpleMessageStream/SimpleMessageStreamProducerExtension.cs

```diff
@@ -196,14 +203,32 @@ internal Task DeliverItem(StreamId streamId, object item, bool fireAndForgetDeli
                             continue;
                     }
 
-                    Task task = remoteConsumer.DeliverItem(subscriptionKvp.Key, immutableItem, null, null);
+                    Task task = DeliverToRemote(remoteConsumer, streamId, subscriptionKvp.Key, immutableItem);
                     if (fireAndForgetDelivery) task.Ignore();
                     else tasks.Add(task);
                 }
                 // If there's no subscriber, presumably we just drop the item on the floor
                 return fireAndForgetDelivery ? TaskDone.Done : Task.WhenAll(tasks);
             }
 
+            private async Task DeliverToRemote(IStreamConsumerExtension remoteConsumer, StreamId streamId, GuidId subscriptionId, Immutable<object> item)
+            {
+                try
+                {
+                    await remoteConsumer.DeliverItem(subscriptionId, item, null, null);
+                }
+                catch (ClientNotAvailableException)
+                {
+                    Tuple<IStreamConsumerExtension, IStreamFilterPredicateWrapper> discard;
+                    if (consumers.TryRemove(subscriptionId, out discard))
+                    {
+                        streamPubSub.UnregisterConsumer(subscriptionId, streamId, streamId.ProviderName).Ignore();
+                        logger.Warn((int)ErrorCode.Stream_ConsumerIsDead,
+                            "Consumer {0} on stream {1} is no longer active - permanently removing Consumer.", remoteConsumer, streamId);
```

Comment:
"Client Consumer {0} on stream {1} is no longer active - permanently removing this Client Consumer and unregisterjng from PubSub".